### PR TITLE
First attempt to try it out.

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: src/main/java/org/opentestsystem/rdw/reporting/controller/GroupResourceAssembler.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     * Creates a new {@link ResourceAssemblerSupport} using the given controller class and resource type.
+     */
+    public GroupResourceAssembler() {
+        super(GroupController.class, GroupResource.class);
+    }
+
+    @Override
+    public GroupResource toResource(Group group) {
+        final List<Link> links = newArrayList();
+        links.add(linkTo(methodOn(GroupController.class).getStudents(group.getId())).withRel("students"));
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/controller/GroupResourceAssembler.java]: no self link?

File: src/main/java/org/opentestsystem/rdw/reporting/service/GroupService.java [hunk 2]
```diff
@@ -2,17 +2,25 @@
...
 
 /**
  * Interface for managing {@link Group}s
  */
 public interface GroupService {
     /**
-     * Find all {@Group}s for the given {@link User}
+     * Find all {@Group}s for the current {@link User}
      *
-     * @param user the {@link User}
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/service/GroupService.java]: Not sure if it is an artifact of the review tool but it looks like you took documentation out for the user parameter, but the user parameter is still there.

File: src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
-		this.repository = checkNotNull(repository);
-	}
+    private final TranslationRepository repository;
 
-	public Set<Translation> getTranslations() {
-		return repository.getTranslations();
-	}
+    @Autowired
+    public TranslationServiceImpl(TranslationRepository repository) {
+        this.repository = checkNotNull(repository);
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java]: thought we weren't doing the checkNotNull thing

File: src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
-	public TranslationServiceImpl(@NotNull TranslationRepository repository) {
-		this.repository = checkNotNull(repository);
-	}
+    private final TranslationRepository repository;
 
-	public Set<Translation> getTranslations() {
-		return repository.getTranslations();
-	}
+    @Autowired
+    public TranslationServiceImpl(TranslationRepository repository) {
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java]: nit: final for all params?

File: src/main/java/org/opentestsystem/rdw/reporting/repository/impl/StudentRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.stereotype.Repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Repository
+public class StudentRepositoryImpl implements StudentRepository {
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Value("${sql.group.student.findAll}")
```
Comment (malaffoon) [inline src/main/java/org/opentestsystem/rdw/reporting/repository/impl/StudentRepositoryImpl.java]: hmm, wondering about the name of the yml entry ... should it be sql.group.student.findAll or sql.group.findAllStudents? Are there going to be any other sql.group.student.* queries?

File: src/main/java/org/opentestsystem/rdw/reporting/controller/GroupResourceAssembler.java (hunk 1 above)
Comment (agorina) [inline src/main/java/org/opentestsystem/rdw/reporting/controller/GroupResourceAssembler.java]: Not yet, I am not sure what it is going to be after all the changes. So for now I left enough to prototype the flow.

File: src/main/java/org/opentestsystem/rdw/reporting/service/GroupService.java (hunk 2 above)
Comment (agorina) [inline src/main/java/org/opentestsystem/rdw/reporting/service/GroupService.java]: Oops, adding it back.

File: src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java (hunk 3 above)
Comment (agorina) [inline src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java]: Will clean it up, I just reformatted this file and did not notice it.

File: src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java (hunk 4 above)
Comment (agorina) [inline src/main/java/org/opentestsystem/rdw/reporting/service/impl/TranslationServiceImpl.java]: Ditto.

File: src/main/java/org/opentestsystem/rdw/reporting/repository/impl/StudentRepositoryImpl.java (hunk 5 above)
Comment (agorina) [inline src/main/java/org/opentestsystem/rdw/reporting/repository/impl/StudentRepositoryImpl.java]: I agree, but this query will go away - so I will change it when we know what it is.

**Pull request merged**